### PR TITLE
HW10

### PR DESCRIPTION
**Pull request by ValeriL**

### CODE REVIEW

File: Homework/homework10/task.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import json
+import re
+from datetime import datetime
+from typing import Dict, List, Tuple
+
+import aiohttp
+
+from bs4 import BeautifulSoup
+
+import requests
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:11]: `aiohttp` and `requests` doing the same thing.    In general, it is not recommended to use them simultaneously.

File: Homework/homework10/task.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+
+def get_company_name(soup) -> str:
+    """Get company name from the page."""
+    return soup.find("span", {"class": "price-section__label"}).text.strip()
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:16]: What is soup?  Does this name represent this variable?

File: Homework/homework10/task.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+def write_json(key: str, data: List[Dict]) -> None:
+    """Write the selected and formatted information about companies into json file."""
+    sorted_data = sort_by_key(data, key)
+    top_data = []
+    for company in sorted_data:
+        top_data.append(
+            {"code": company["code"], "name": company["name"], f"{key}": company[key]}
+        )
+    with open("top_10_" + key + ".json", "w+") as f:
+        json.dump(top_data, f)
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:138]: Make it pretty:
```python
json.dump(top_data, f, indent=4)
```

File: Homework/pytest.ini [hunk 4]
```diff
@@ -1,2 +1,4 @@
 [pytest]
 addopts = --doctest-modules --doctest-continue-on-failure
+filterwarnings =
+    ignore:"@coroutine" decorator is deprecated since Python 3.8, use "async def" instead:DeprecationWarning
```
Comment (Cjkjvfnby) [inline Homework/pytest.ini:4]: Why do you ignore it?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
You can always split tasks into multiple files. One responsible for fetching data, one for parsing, and the last one for combining this together.  Optional.

File: Homework/homework10/task.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+def get_profit(low_high_values: Tuple[float, float]) -> float:
+    """Get a value of profit if we buy the stock at 52 Week Low price and sell at 52 Week High price."""
+    low, high = low_high_values
+    return round((high - low) / low * 100, 2)
+
+
+def sort_by_key(data: List[Dict], value: str) -> List[Dict]:
+    """Sort the company info by given key value."""
+    return sorted(data, key=lambda x: x[value], reverse=True)[0:10]
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:63]: https://docs.python.org/3/library/heapq.html?highlight=nlargest#heapq.nlargest

File: Homework/homework10/task.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    day = current_datetime.day
+    formatted_day = day if len(str(day)) > 1 else "0" + str(day)
+    bank_url = f"http://www.cbr.ru/scripts/XML_daily.asp?date_req={formatted_day}/{formatted_month}/{year}"
+    currency_page_soup = BeautifulSoup(await fetch_html(bank_url), "lxml")
+    currency = (
+        currency_page_soup.find("valute", {"id": "R01235"})
+        .find("value")
+        .text.replace(",", ".")
+    )
+    return float(currency)
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:84]: Should we round it too?

File: Homework/homework10/task.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+async def get_company_data(company: List, currency) -> Dict:
+    """Get the information about the company."""
+    url = "https://markets.businessinsider.com" + company["href"]
+    company_page_soup = BeautifulSoup(await fetch_html(url), "lxml")
+    return {
+        "name": get_company_name(company_page_soup),
+        "code": get_company_code(company_page_soup),
+        "price": get_stock_price(company_page_soup, currency),
+        "pe_ratio": get_pe_ratio(company_page_soup),
```
Comment (Cjkjvfnby) [inline Homework/homework10/task.py:102]: what is `pe ratio`?

**Review by Cjkjvfnby — APPROVED** [review]

File: Homework/homework10/task.py (hunk 6 above)
Comment (ValeriL) [reply Homework/homework10/task.py:84]: But we don't need to show it anywhere, its used only for calculating stock price which we rounds

File: Homework/homework10/task.py (hunk 7 above)
Comment (ValeriL) [reply Homework/homework10/task.py:102]: P/E ratio

Comment (Cjkjvfnby) [reply Homework/homework10/task.py:102]: Do you mean `price_to_earnings_ratio`?

**Pull request merged**